src/main/scala/cromwell/engine/workflow/ExecutionStoreKey.scala

```diff
@@ -0,0 +1,66 @@
+package cromwell.engine.workflow
+
+import cromwell.binding._
+import cromwell.engine.ExecutionStatus
+import cromwell.engine.workflow.WorkflowActor.ExecutionStore
+
+sealed trait ExecutionStoreKey {
+  val scope: Scope
+
+  val index: Option[Int]
+
+  val parent: Option[ExecutionStoreKey]
+}
+
+
+case class CallKey(override val scope: Call,
+                   override val index: Option[Int],
+                   override val parent: Option[ExecutionStoreKey])
+  extends ExecutionStoreKey
+
+
+case class ScatterKey(override val scope: Scatter,
+                      override val index: Option[Int],
+                      //override val status: ExecutionStatus,
+                      override val parent: Option[ExecutionStoreKey])
+  extends ExecutionStoreKey {
+
+  /**
+   * Creates a sub-ExecutionStore with Starting entries for each of the scoped children.
+   * @param count Number of ways to scatter the children.
+   * @return ExecutionStore of scattered children.
+   */
+  def populate(count: Int): ExecutionStore = {
+    val keys = for {
+      childScope <- this.scope.children
+      collector = collectorForChild(childScope, count)
+      entry <- collector +: collector.keys
+    } yield entry
+
+    (keys map {
+      _ -> ExecutionStatus.Starting
+    }).toMap
+  }
+
```

Comment:
Off the top of my head, under our current principle: "The database's job is to tell you about running jobs", `Collector` would never be directly stored in the database. So it would be the latter, "an issue how will the `keys` attribute be populated on workflow restart".

I therefore imagined that, at startup, when some array of `Call` instances re-instantiated from the db, the array should have a corresponding `Collector` object created at the same time.

Happy to refactor in this PR, but do want to focus on bugs/blockers/issues from getting it merged into the `scatter-gather` branch, since soon this won't be going into `develop` right away.